app/controllers/comments.controller.js

```diff
@@ -0,0 +1,18 @@
+const {fetchCommentsByArticleId} = require('../models/comments.model')
+
+
+
+/// invalid id error
+
+
+exports.getCommentsByArticleId = (req,res,next) => {
+    const article_id = req.params.article_id;
+    fetchCommentsByArticleId(article_id).then((comments) => {
+        if (comments.length === 0){
+            next(comments)
```

Comment:
We should be allowing the catch block to pick up any errors that occur during the promise chain and pass it to our middleware functions, any logic we write to potentially throw an error should be in our model, not the controller. Have a look at Promise.reject to help.